tests/test_cli.py

```
@@ -258,6 +265,60 @@ def test_clean(
+    tmp_dir_cwd: Path,
+    logfire_credentials: LogfireCredentials,
+    capsys: pytest.CaptureFixture[str],
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """A crafted credentials file can't smuggle escape sequences into the user's terminal."""
+    monkeypatch.setattr(sys, 'stdin', io.StringIO('y'))
+
+    logfire_credentials.project_name = 'my-\x1b[2Kproject'
+    logfire_credentials.project_url = 'https://dashboard.logfire.dev\x1b]0;pwned\x07'
+    logfire_credentials.write_creds_file(tmp_dir_cwd)
+    main(shlex.split(f'clean --data-dir {str(tmp_dir_cwd)}'))
```

Ruff (0.16.1)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)